packages/next/server/next-dev-server.ts

```diff
@@ -330,24 +330,35 @@ export default class DevServer extends Server {
   }
 
   generateRoutes() {
-    const routes = super.generateRoutes()
+    const {
+      routes,
+      fsRoutes,
+      catchAllRoute,
+      dynamicRoutes,
+      pageChecker,
+    } = super.generateRoutes()
 
     // In development we expose all compiled files for react-error-overlay's line show feature
     // We use unshift so that we're sure the routes is defined before Next's default routes
-    routes.unshift({
-      match: route('/_next/development/:path*'),
-      type: 'route',
-      name: '_next/development catchall',
-      fn: async (req, res, params) => {
-        const p = join(this.distDir, ...(params.path || []))
-        await this.serveStatic(req, res, p)
-        return {
-          finished: true,
-        }
+    const devRoutes: Route[] = [
+      {
+        match: route('/_next/development/:path*'),
+        type: 'route',
+        name: '_next/development catchall',
+        fn: async (req, res, params) => {
+          const p = join(this.distDir, ...(params.path || []))
+          await this.serveStatic(req, res, p)
+          return {
+            finished: true,
+          }
+        },
       },
-    })
+    ]
+
+    routes.unshift(...devRoutes)
```

Comment:
Why is this in both routes and fsRoutes? AFAIK this should be strictly fsRoutes.

Reply:
It's in both right now since `fsRoutes` is only used with `check: true` and `routes` contains the routes that are always checked. Could keep them separate and run a check over `fsRoutes` before the normal `routes`

Reply:
Updated to keep them separate instead of merging them together